synapse/federation/federation_base.py

```diff
@@ -238,6 +238,10 @@ def event_from_pdu_json(
     # origin, etc etc)
     assert_params_in_dict(pdu_json, ("type", "depth"))
 
+    # Strip any unauthorized values from "unsigned" if they exist
+    if "unsigned" in pdu_json and pdu_json["unsigned"] != {}:
+        pdu_json = _strip_unsigned_values(pdu_json)
```

Comment:
it's worth noting that, because `_strip_unsigned_values` modifies the actual dict that `pdu_json` refers to, this is the same as:

```suggestion
        _strip_unsigned_values(pdu_json)
```

Indeed, I think it's confusing to have `_strip_unsigned_values` return the dict at all, since it gives the incorrect impression that it returns a *different* dict, rather than modifying the original.

Reply:
That makes sense, thank you for pointing that out!